### ISSUE TITLE
Menu Bar - Strange keyboard navigation used

### ISSUE THREAD

**Issue by audio2000** [audacity/audacity#10004]
### Bug description

`Menu Bar` does not follow standard navigational functionality when using keyboard ARROW keys

### Steps to reproduce

1. Click on `Menu Bar` category to open drop down
2. Now try common navigation using your UP DOWN LEFT RIGHT ARROW keys

### Expected behavior

To use ALT+letter or click on `Menu Bar` categories with mouse it opens drop downs.
Then be able to experience standard navigational behavior when using ARROW keys

LEFT ARROW / RIGHT ARROW = would open the drop down for the category to the left/right
LEFT ARROW / RIGHT ARROW FLYOUTS = would open / close flyout
UP ARROW / DOWN ARROW = navigate up/down

### Actual behavior

LEFT ARROW = closes the drop down
RIGHT ARROW = navigates highlight cursor down (strange)
RIGHT ARROW ON FLYOUT = opens flyout

If LEFT ARROW is used to close the drop down, you then seen white rectangle (if using dark theme) around the categories name.
This then allows LEFT / RIGHT ARROWS to go left / right to other category names but does not open the drop down

### Audacity Version

Audacity 4 Alpha2 - Official AppImage

### Operating system

Linux

### Additional context

Hopefully this functionality is a bug in Audacity that can be fixed and not a decision made by QT that cannot be.

RELATED POST
Another standard behavior that is missing is to be able to close drop downs using ESCAPE key which I made a post about here if anyone is interested
https://github.com/audacity/audacity/issues/10003

**Comment by DavidBailes** [comment]
This is an issue with Audacity, and not Qt.
In the program musescore studio, which is developed by the same people, and also uses Qt, the keyboard navigation is as expected.

**Comment by dozzzzer** [comment]
What I'm seeing on Windows/Linux:

<kbd>↑</kbd> highlights the last menu item 
<kbd>↓</kbd> highlights the second menu item if it exists; otherwise, the first item
<kbd>→</kbd> highlights the second menu item if it exists; otherwise, the first item
<kbd>←</kbd> closes the dropdown menu

What I'm seeing on MacOS:

<kbd>↑</kbd> highlights the last menu item 
<kbd>↓</kbd> highlights the first menu item
<kbd>→</kbd> opens the dropdown for the category to the right
<kbd>←</kbd> opens the dropdown for the category to the left

We only need to align the Windows/Linux behavior with MacOS.

Also, on my end, the current version of Musescore doesn't seem to match the expected behavior:

<kbd>↑</kbd> highlights the last menu item 
<kbd>↓</kbd> highlights the second menu item if it exists; otherwise, the first item
<kbd>→</kbd> opens the dropdown for the leftmost category and 
<kbd>←</kbd> opens the dropdown for the category to the left

**Comment by DavidBailes** [comment]
> Also, on my end, the current version of Musescore doesn't seem to match the expected behavior:
> 
> ↑ highlights the last menu item ↓ highlights the second menu item if it exists; otherwise, the first item → opens the dropdown for the leftmost category and ← opens the dropdown for the category to the left

@dozzzzer . I hadn't noticed that right arrow doesn't have the correct behaviour.
However, for down arrow this is the expected behaviour when using the WinUI 3 framework, which is Microsoft's latest framework for developing desktop apps.
An app is provided which illustrates behaviour of apps developed using this framework: https://aka.ms/winuigallery
In this app, in the menus and toolbars category, there's a menubar.

**Comment by dozzzzer** [comment]
This is still reproducible
